### PR TITLE
Added support for CSN A2L with firmware 2.168

### PR DESCRIPTION
**Pull request by greg-elmi**
Creating separate subclass for new firmware. It resolves issue  #18 of printing leftover characters as new firmware doesn't support darkness manipulation commands.

I did not go through whole new manual to check for other changes. I'll be adding them on the go (or someone else can)

### CODE REVIEW

**Comment by tannewt** [comment]
Thanks @greg-elmi! How did you get a printer with newer firmware? @ladyada are we now selling printers with newer firmware?

**Comment by ladyada** [comment]
not that we were informed of

**Comment by greg-elmi** [comment]
I did not buy it from Adafruit but from Cashino's distributor.
There's also possibility that A2 has different firmware than A2L (The one i bought. The difference is USB port option), but I did not delve into that

File: adafruit_thermal_printer/thermal_printer_264.py [hunk 1]
```diff
@@ -28,7 +28,8 @@
 versions and care must be taken to select the appropriate module inside this
 package for your firmware printer:
 
-* thermal_printer = The latest printers with firmware version 2.68+
+* thermal_printer_2164 = Printers with firmware version 2.168+.
```
Comment (tannewt) [inline adafruit_thermal_printer/thermal_printer_264.py:31]:
```suggestion
* thermal_printer_2168 = Printers with firmware version 2.168+.
```

File: adafruit_thermal_printer/thermal_printer_legacy.py [hunk 2]
```diff
@@ -28,7 +28,8 @@
 versions and care must be taken to select the appropriate module inside this
 package for your firmware printer:
 
-* thermal_printer = The latest printers with firmware version 2.68+
+* thermal_printer_2164 = Printers with firmware version 2.168+.
```
Comment (tannewt) [inline adafruit_thermal_printer/thermal_printer_legacy.py:31]:
```suggestion
* thermal_printer_2168 = Printers with firmware version 2.168+.
```

File: adafruit_thermal_printer/thermal_printer_2168.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+"""
+`adafruit_thermal_printer.thermal_printer_2168.ThermalPrinter`
+==============================================================
+
+Thermal printer control module built to work with small serial thermal
+receipt printers.  Note that these printers have many different firmware
+versions and care must be taken to select the appropriate module inside this
+package for your firmware printer:
+
+* thermal_printer_2164 = Printers with firmware version 2.168+.
```
Comment (tannewt) [inline adafruit_thermal_printer/thermal_printer_2168.py:32]:
```suggestion
* thermal_printer_2168 = Printers with firmware version 2.168+.
```

File: adafruit_thermal_printer/thermal_printer.py [hunk 4]
```diff
@@ -386,7 +387,8 @@ def set_defaults(self):
         self.size = SIZE_SMALL
         self.underline = None
         self.inverse = False
-        self.upside_down = False
+        self.upside_down = False  # wg. dokumentacji powinno działać ale nie działa
+        self.up_down_mode = True  # zamiast powyższego zaiplementowałem to   <<
```
Comment (tannewt) [inline adafruit_thermal_printer/thermal_printer.py:391]: Mind commenting these in English? Most folks we work with (including myself) speak English.

**Review by tannewt — CHANGES_REQUESTED** [review]
Just a couple minor things. Thanks for adding this support!

File: adafruit_thermal_printer/thermal_printer.py (hunk 4 above)
Comment (greg-elmi) [reply adafruit_thermal_printer/thermal_printer.py:391]: Yes. Sorry. I worked at this part like 2 months ago and didn't pay attention to it when I was preparing for commit

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**